File: datasets/C++/Mistral/DataStructures/medium1.cpp
#include <iostream>
#include <vector>
#include <string>
#include <map>

using namespace std;

vector<int> matchingStrings(vector<string>& stringList, vector<string>& queries) {
    map<string, int> frequencyMap;
    for (const string& str : stringList) {
        frequencyMap[str]++;
    }

    vector<int> result;
    for (const string& query : queries) {
        result.push_back(frequencyMap[query]);
    }

    return result;
}

int main() {
    int n;
    cout << "Enter the size of stringList: ";
    cin >> n;
    cin.ignore(); // Ignore the newline character

    vector<string> stringList(n);
    for (int i = 0; i < n; ++i) {
        getline(cin, stringList[i]);
    }

    int q;
    cout << "Enter the size of queries: ";
    cin >> q;
    cin.ignore(); // Ignore the newline character

    vector<string> queries(q);
    for (int i = 0; i < q; ++i) {
        getline(cin, queries[i]);
    }

    vector<int> result = matchingStrings(stringList, queries);
    for (int count : result) {
        cout << count << " ";
    }
    cout << endl;

    return 0;
}

File: datasets/Python/Mistral/DataStructures/medium1.py
def matching_strings(string_list, queries):
    result = []
    for query in queries:
        count = string_list.count(query)
        result.append(count)
    return result

# Reading input
n = int(input("Enter the size of stringList: "))
string_list = [input().strip() for _ in range(n)]
q = int(input("Enter the size of queries: "))
queries = [input().strip() for _ in range(q)]

# Getting the result
result = matching_strings(string_list, queries)
print(result)
